pkg/install/resource/tm_deployment.go

```diff
@@ -0,0 +1,183 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/datawire/dlib/dlog"
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+	"github.com/telepresenceio/telepresence/v2/pkg/install"
+)
+
+const managerLicenseName = "systema-license"
+
+type tmDeployment struct {
+	found *kates.Deployment
+}
+
+var TrafficManagerDeployment Instance = &tmDeployment{}
+
+func (ri *tmDeployment) deployment(ctx context.Context) *kates.Deployment {
+	dep := new(kates.Deployment)
+	dep.TypeMeta = kates.TypeMeta{
+		Kind: "Deployment",
+	}
+	sc := getScope(ctx)
+	dep.ObjectMeta = kates.ObjectMeta{
+		Namespace: sc.namespace,
+		Name:      install.ManagerAppName,
+		Labels:    sc.tmSelector,
+	}
+	return dep
+}
+
+func (ri *tmDeployment) desiredDeployment(ctx context.Context) *kates.Deployment {
+	replicas := int32(1)
+
+	sc := getScope(ctx)
+	var containerEnv = []corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "info"},
+		{Name: "SYSTEMA_HOST", Value: sc.env.SystemAHost},
+		{Name: "SYSTEMA_PORT", Value: sc.env.SystemAPort},
+		{Name: "CLUSTER_ID", Value: sc.clusterID},
+		{Name: "TELEPRESENCE_REGISTRY", Value: sc.env.Registry},
+
+		// Manager needs to know its own namespace so that it can propagate that when
+		// to agents when injecting them
+		{
+			Name: "MANAGER_NAMESPACE",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					FieldPath: "metadata.namespace",
+				},
+			},
+		},
+	}
+	if sc.env.AgentImage != "" {
+		containerEnv = append(containerEnv, corev1.EnvVar{Name: "TELEPRESENCE_AGENT_IMAGE", Value: sc.env.AgentImage})
+	}
+
+	optional := true
+	volumes := []corev1.Volume{
+		{
+			Name: "license",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: managerLicenseName,
+					Optional:   &optional,
+				},
+			},
+		},
+		{
+			Name: "tls",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: install.MutatorWebhookTLSName,
+					Optional:   &optional,
+				},
+			},
+		},
+	}
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "license",
+			ReadOnly:  true,
+			MountPath: "/home/telepresence/",
+		},
+		{
+			Name:      "tls",
+			ReadOnly:  true,
+			MountPath: "/var/run/secrets/tls",
+		},
```

Comment:
You added this in "Move install, update, and delete of traffic-manager to resource package" but probably didn't mean to add it until "Add a mutator webhook service to the traffic-manager".  You don't need to change it, but I got thrown through a loop for a minute.